lcls-twincat-motion/Library/POUs/Motion/PMPS/FB_Standard_PMPSDB.TcPOU

```diff
@@ -62,9 +66,30 @@ IF rtEnable.Q OR rtRefresh.Q THEN
     bExecute := TRUE;
 END_IF
 
+IF sDirectory = '' THEN
+    // Check OS for default directory
+    fbIPCReg(bExecute:=TRUE);
+    fbCheckOS(
+        bExecute:=NOT fbIPCReg.bBusy,
+        eParameterKey:=E_IPCDiag_ParameterKey.OS_Name,
+        fbRegister:=fbIPCReg,
+    );
+    IF fbCheckOS.bValid THEN
```

Comment:
I noticed this includes a check for error, but if there is an error, we won't be flagged.

Reply:
Good catch. You're right, as-written if we have an error the JSON file will silently never load and we might not notice an issue until we check the PMPS diagnostic.

This warrants some revision.

Reply:
Some options that I'm tossing around right now:

- If there's an error, try again? (a few times?)
- Go back to the old default?
- Give an error log message?
- Fast fault?

Needs some thought

Reply:
I'm going to opt for 3 tries followed by going back to the old default
This gives us some wiggle room for first-cycle-bs and will give us some normal error messages if the old default was wrong and if the old default was correct then the PLC will load values as normal.

Reply:
I think this means, if the OS isn't solved and a twincat PLC gets the default directory, fbPmpsFileReader will throw a directory not found error.